### PR TITLE
Removed RouteRefreshOptions#enabled in favor of respecting RouteOptions#enableRefresh

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Refs https://github.com/mapbox/mapbox-navigation-android/pull/4526. This removes the now redundant `RouteRefreshOptions#enabled`. Since `RouteOptions#enableRefresh` is exposed, we should respect that instead.

This PR also fixes a flaky route refresh instrumentation test which was always timing out because the idling calls were sequenced incorrectly.

### Changelog

```
<changelog>:warning: Removed `RouteRefreshOptions#enabled`. To enable the route refresh feature (that updates congestion and other annotations on the currently active route) you need to enable it via `RouteOptions#enableRefresh` when making a route request.</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt [hunk 1]
```diff
@@ -45,7 +41,7 @@ internal class RouteRefreshController(
      */
     fun restart() {
         stop()
-        if (routeRefreshOptions.enabled) {
+        if (tripSession.route?.routeOptions()?.enableRefresh() == true) {
```
Comment (korshaknn) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt:44]: we may add extension
`fun DirectionsRoute?.enableRefresh() : Boolean`

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt [hunk 2]
```diff
@@ -65,7 +61,7 @@ internal class RouteRefreshController(
 
     private fun refreshRoute() {
         val route = tripSession.route
-            ?.takeIf { it.routeOptions().supportsRouteRefresh() }
+            ?.takeIf { it.routeOptions()?.enableRefresh() == true }
```
Comment (korshaknn) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt:64]: the same extension

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt (hunk 1 above)
Comment (LukasPaczos) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt:44]: I don't think that's worth it, the same argument could be made for all other `RouteOptions`.

**Comment by Guardiola31337** [comment]
Noting that there are some failing tests https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/12920/workflows/a011f341-49c1-4cf4-8e5b-a18252e81d14/jobs/53237

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 3]
```diff
@@ -482,8 +482,8 @@ class MapboxNavigation(
     fun setRoutes(routes: List<DirectionsRoute>) {
         rerouteController?.interrupt()
         routeAlternativesController.interrupt()
-        routeRefreshController.restart()
         directionsSession.routes = routes
+        routeRefreshController.restart()
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:486]: Does changing the order here matters? 🤔 

Edited: Yeah, it seems so 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/b3aca80bcdc03cca189da2ed97acecfee1cd4295/libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt#L295-L300

**Review by Guardiola31337 — APPROVED** [review]

**Pull request merged**